smart/model.py

```diff
@@ -938,6 +938,7 @@ def _init_4_7_set_initial_conditions(self):
                     self.dolfin_set_function_values(
                         species, ukey, species.initial_condition_expression
                     )
+            species.sol = species.u["u"]  # assign solution to species.sol for easy access
```

Comment:
I wouldn't do this in this way. I would do something like
```python
@property
def sol(self):
    return self.u["u"]
```
in the species class.

Reply:
Or you could also add some error handling, e.g do something like
```python
@property
def sol(self):
    if not hasattr(self, "u") or "u" not in self.u:
        raise RunTimeError("Solution does not exist. Please run a simulation first")
    return self.u["u"]
```
or something along those lines